### PR TITLE
[skip ci] Fixing camera overlay stacking adding to respect unity general reference restrictions

### PR DESCRIPTION
**Pull request by lukaschod**
### Checklist for PR maker
- [x] Have you added a backport label (if needed)? For example, the `need-backport-*` label. After you backport the PR, the label changes to `backported-*`.
- [x] Have you updated the changelog? Each package has a `CHANGELOG.md` file.
- [ ] Have you updated or added the documentation for your PR? When you add a new feature, change a property name, or change the behavior of a feature, it's best practice to include related documentation changes in the same PR.
- [ ] Have you added a graphic test for your PR (if needed)? When you add a new feature, or discover a bug that tests don't cover, please add a graphic test.

---
### Purpose of this PR
Fix for case https://fogbugz.unity3d.com/f/cases/1240788/.

---
### Testing status

**Manual Tests**: What did you do?
- [x] Opened test project + Run graphic tests locally
- [x] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo + Prefab overrides + Alignment in Preset
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in developer mode, you have a button at end of resources that check the paths)
- Other: 

**Automated Tests**: Nothing as this is editor only fix.

**Yamato**: https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/universal%252Fbugfix%252Fcamera-stack-adding

---
### Comments to QA
This PR only effects camera stack add button. I included two new restrictions:
- Prefab can not reference scene cameras.
- Scene camera can not reference other scene camera.

I need sanity test on camera stack add button.

### CODE REVIEW

File: com.unity.render-pipelines.universal/Editor/UniversalRenderPipelineCameraEditor.cs [hunk 1]
```diff
@@ -311,12 +311,55 @@ void DrawElementCallback(Rect rect, int index, bool isActive, bool isFocused)
             }
         }
 
+        // Modified version of StageHandle.FindComponentsOfType<T>()
+        // This version more closely represents unity object referencing restrictions.
+        // I added these restrictions:
+        // - Can not reference scene object outside scene
+        // - Can not reference cross scenes
+        // - Can reference child objects if it is prefab
+        Camera[] FindCamerasToReference(GameObject gameObject)
```
Comment (eh-unity) [inline com.unity.render-pipelines.universal/Editor/UniversalRenderPipelineCameraEditor.cs:320]: Based on above comment. Shouldn't this then be `FindTypeToReference<Camera>(GameObject ... )`?
Also is this really a correct place? Shouldn't we have a common implementation?

File: com.unity.render-pipelines.universal/Editor/UniversalRenderPipelineCameraEditor.cs [hunk 2]
```diff
@@ -311,12 +311,55 @@ void DrawElementCallback(Rect rect, int index, bool isActive, bool isFocused)
...
+        {
+            var scene = gameObject.scene;
+
+            var notInScene = EditorUtility.IsPersistent(camera) && !scene.IsValid();
```
Comment (eh-unity) [inline com.unity.render-pipelines.universal/Editor/UniversalRenderPipelineCameraEditor.cs:324]: Could we have this as a positive variable `inScene` like the others. I find that more readable if there ever is negations and it's consistent with the other bools.

Comment (lukaschod) [reply com.unity.render-pipelines.universal/Editor/UniversalRenderPipelineCameraEditor.cs]: k done it

File: com.unity.render-pipelines.universal/Editor/UniversalRenderPipelineCameraEditor.cs (hunk 1 above)
Comment (lukaschod) [reply com.unity.render-pipelines.universal/Editor/UniversalRenderPipelineCameraEditor.cs:320]: Well I was also considering this, but I decided not to do it for few reasons:
- This function is mostly tailored for finding cameras for camera stacking criteria (etc. we can not include camera prefab in scene).
- This function is only used once.

Now about template point. As this function is mainly tailored for camera stacking adding, for this reason there is no point in my opinion to make it work with other types.

**Pull request merged**